core/templates/domain/admin/beam-job-run.model.ts

```diff
@@ -0,0 +1,180 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Domain object for an instance of an Apache Beam job.
+ */
+
+export type BeamJobRunState = (
```

Comment:
Hmm I think I like it more here since it provides a strongly-typed enum (code fails to compile if it uses a wrong value). I don't think using constants would give us that same power.

Reply:
We will be introducing protobuf for constants soon and that will enable us to use enums, so maybe add a TODO comment to move this to constants after this is done.